From 40fc6b25c9dcad6bbcdf4ac2e56aa39bb5c68d4f Mon Sep 17 00:00:00 2001
From: Robin Gong <yibin.gong@nxp.com>
Date: Tue, 18 Jul 2017 17:01:17 +0800
Subject: [PATCH 2179/5242] MLK-15305-2: dma: imx-sdma: force to load context
 in sdma_config

commit  eed183baf5e2dba01f069e26315f87769b447508 from
https://source.codeaurora.org/external/imx/linux-imx.git

Force to load context in sdma_config whatever context loaded or not,
since some configuration may change when the upper driver call sdma_config
such as bus width.

Signed-off-by: Robin Gong <yibin.gong@nxp.com>
(cherry picked from commit ee8930b657af0c9ce2cfb1a521530c7d31016675)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/dma/imx-sdma.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/dma/imx-sdma.c b/drivers/dma/imx-sdma.c
index c5f5ac9..877cb8c 100644
--- a/drivers/dma/imx-sdma.c
+++ b/drivers/dma/imx-sdma.c
@@ -1207,6 +1207,8 @@ static int sdma_config_channel(struct dma_chan *chan)
 		sdmac->watermark_level = 0; /* FIXME: M3_BASE_ADDRESS */
 	}
 
+	sdmac->context_loaded = false;
+
 	ret = sdma_load_context(sdmac);
 
 	return ret;
-- 
1.7.9.5

